debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -118,6 +118,13 @@ public R apply(final R record) {
             return record;
         }
 
+        if (record.valueSchema() == null ||
+                record.valueSchema().name() == null ||
+                !record.valueSchema().name().endsWith(ENVELOPE_SCHEMA_NAME_SUFFIX)) {
+            LOGGER.warn("Expected Envelope for transformation, passing it unchanged");
```

Comment:
Is it actually right to warn about it? It's expected after all, thinking of heartbeat or schema messages. Rather `debug()` perhaps?

Reply:
@gunnarmorling IMHO the original idea was that there will be a sequence of SMTs and this SMT will receive only those that could be processed, so the message should be an exception rather than rule.

Reply:
This also affects outbox and mongo right? Maybe it's time to extract this functionality?

Reply:
@gunnarmorling @renatomefi I wouldn't change it in this PR and instead open a new issue with it. Let's keep the scope as small as possible.

Reply:
Ok, assuming the SMT is mostly applied on the sink side, schema messages and heartbeats indeed are rather unexpected. So the warning seems ok.